src/atomWithHash.ts

```diff
@@ -51,7 +57,7 @@ export function atomWithHash<Value>(
   if (typeof setHashOption === 'function') {
     setHash = setHashOption;
   }
-  const strAtom = atom<string | null>(null);
+  const strAtom = atom(initialValueInHash);
```

Comment:
On second thought, I might prefer this:
```suggestion
  const strAtom = atom(
    (typeof window === 'undefined' || !window.location)
      ? null
      : new URLSearchParams(window.location.hash.slice(1)).get(key)
  );
```
We might be able to refactor some more with the `onMount` below.

Like,
```js
  const isLocationAvailable = typeof window !== 'undefined' && !!window.location;
  const strAtom = atom(
    isLocationAvailable
      ? new URLSearchParams(window.location.hash.slice(1)).get(key)
      : null
  );
  strAtom.onMount = (setAtom) => {
    if (!isLocationAvailable) {
      return;
    }
    const callback = () =>
      setAtom(new URLSearchParams(window.location.hash.slice(1)).get(key));
    const unsubscribe = subscribe(callback);
    callback();
    return unsubscribe;
  };
```

Reply:
Tests are passing, and that is a nicer, more 'contained' solution. I like it! 👍